src/s3/commands/uploadFile.ts

```diff
@@ -114,11 +148,221 @@ async function uploadWithProgress({
+        return bucket && bucket.Name
+    }) as S3.Bucket[]
+
+    
+    const createNewBucket: BucketQuickPickItem = {
+        label: localize('AWS.command.s3.createBucket',"Create new bucket"),
+        bucket: undefined
+    }
+    const bucketItems: BucketQuickPickItem[] = s3Buckets.map(bucket => {
+        return { 
+            label: bucket.Name!,
+            bucket
+        }
+    })
+
+    const picker = createQuickPick({
+        options: {
+            canPickMany: false,
+            ignoreFocusOut: true,
+            title: localize('AWS.message.selectBucket','Select S3 bucket to upload to'),//localize('AWS.lambda.upload.title', 'Select Upload Type'),
+            step: 2,
+            totalSteps: 2,
+        },
+        items: [...bucketItems, createNewBucket],
+        buttons: [vscode.QuickInputButtons.Back]
+    })
+    const response = verifySinglePickerOutput(await promptUser({
+        picker: picker,
+        onDidTriggerButton: (button, resolve, reject) => {
+            if (button === vscode.QuickInputButtons.Back) {
+                resolve([{
+                    label: "back",
+                    bucket: undefined
+                }])
+            }
+
+        },
+    })) 
+
+    if (!response) {
+        return undefined
+    }
+
+    if (!response.bucket) {
+        if (response.label === 'back') {
+            uploadFilePaletteCommand()
+        }
+        if (response.label === 'Create new bucket') {
+            getLogger().debug('CreateBucket called for: %O', s3client)
+
+            const bucketName = await window.showInputBox({
+                prompt: localize('AWS.s3.createBucket.prompt', 'Enter a new bucket name'),
+                placeHolder: localize('AWS.s3.createBucket.placeHolder', 'Bucket Name'),
+                validateInput: validateBucketName,
+            })
+
+            if (!bucketName) {
+                getLogger().info('CreateBucket cancelled')
+                telemetry.recordS3CreateBucket({ result: 'Cancelled' })
+                return undefined
+            }
+
+            getLogger().info(`Creating bucket: ${bucketName}`)
+            try {
+                const bucket = await s3client.createBucket({ bucketName })
+
+                getLogger().info('Created bucket: %O', bucket)
+                window.showInformationMessage(localize('AWS.s3.createBucket.success', 'Created bucket: {0}', bucketName))
+                telemetry.recordS3CreateBucket({ result: 'Succeeded' })
+                return { Name: bucket.bucket.name }
+            } catch (e) {
+                getLogger().error(`Failed to create bucket ${bucketName}: %O`, e)
+                showErrorWithLogs(
+                    localize('AWS.s3.createBucket.error.general', 'Failed to create bucket: {0}', bucketName),
+                    Window.vscode()
+                )
+                telemetry.recordS3CreateBucket({ result: 'Failed' })
+                return undefined
+            }
+        }
```

Comment:
Is this a direct copy/paste? If so, see if you can extract it to avoid duplication.